tests/cmdline/commands/test_data.py

```diff
@@ -310,8 +311,17 @@ def connect_structure_bands(strct):  # pylint: disable=unused-argument
     @classmethod
     def setUpClass(cls):  # pylint: disable=arguments-differ
         super().setUpClass()
+
+        # create a new event loop since the privious one is closed by other test case
```

Comment:
> I think it might be safer to move this to setUp and tearDown so it's done per test case rather than per test case class.

Sure, `setUp` and `tearDown` of `AiidaTestCase` already did the new event loop stuff. The exist of the code here is because of the `cls.ids = cls.create_structure_bands()` which is a `ProcessFunction`,  it requires a not closed event loop.  The `setUpClass` is running before the `setUp`, so I have to manually create a event loop. The other option would be remove the ProcessFunction to somewhere else, I tried but failed.